plugin.audio.pureradio/purelib.py

```diff
@@ -0,0 +1,150 @@
+
+import xbmc
+import xbmcaddon
+import xbmcplugin
+import xbmcgui
+
+import os
+import sys
+
+import urllib.parse
+import urllib.request
+from xml.etree import ElementTree as ET
+
+
+ADDONID      = 'plugin.audio.pureradio'
+ADDON        = xbmcaddon.Addon(ADDONID)
+ADDON_HANDLE = int(sys.argv[1])
+
+HOME         = ADDON.getAddonInfo('path')
+AUDIO_ICON   = "DefaultAudio.png"
+FOLDER_ICON  = "DefaultFolder.png"
+FANART       = os.path.join(HOME,'resources','fanart.jpg')
+
+PLAYSTREAM  = 100
+PODCASTS    = 101
+PLAYPODCAST = 102
+
+URL         = "http://listento.pureradio.one/pureradio.pls"
+PODCASTURL  = 'http://www.spreaker.com/show/1757189/episodes/feed'
+
+params = {}
+
+def check_autoPlay():
+    if (ADDON.getSetting('PLAY')=='true'):
+        playStream()    
+
+def playStream():
+    # music item
+    pl = xbmc.PlayList(xbmc.PLAYLIST_MUSIC)
+    # clear playlist
+    pl.clear()
+    # add url item 
+    pl.add(URL)
+    # play the item
+    xbmc.Player().play(pl)
```

Comment:
To use xbmc.Player().play() for a playlist in a plugin requires a bit more work. Direct urls should not be in the playlist. Playlist items need to be of the form plugin://......... with the url as a parameter so that the plugin gets called to play the url. The play routine should make use of  xbmcplugin.setResolvedUrl() to actually play the playable url.

xbmc.Player().play() should not be used to play a url in a plugin. It should be used in a script.